gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -289,6 +272,84 @@ def run(self) -> GtsfmData:
             process_graph_generator.is_image_correspondence = True
         process_graph_generator.save_graph()
 
+        if self.parsed_args.run_low_quality:
+            # TODO(johnwlambert): Check if these are None, otherwise don't override.
+            COMBINATIONS = [(self.parsed_args.num_matched, self.parsed_args.max_frame_lookahead)]
+        else:
+            # Run multiple trials at different (num_matched, max_frame_lookahead) settings.
+            COMBINATIONS = [
+                (0, 5),
+                (0, 10),
+                (5, 0),
+                (5, 5),
+                (5, 10),
+                (10, 0),
+                (10, 5),
+                (10, 10),
+            ]
+
+        experiment_roots = []
+        output_root = self.scene_optimizer.output_root
+        for (num_matched, max_frame_lookahead) in COMBINATIONS:
```

Comment:
There are retrievers which work with only one of num_matched/max_frame_lookahead? We throw value errors which might be wrong. Example for `ImageMatchingRegime.RETRIEVAL`